### DISCUSSION TITLE
Trasforming MD before updating the editor

### DISCUSSION THREAD

**Discussion by ivanminutillo** [Milkdown/milkdown#1086] [Q&A]
hello, and thanks for working on milkdown, is the best experience using a modular wyswyg md editor by far 🙂 
I'm currently stuck in something that I am sure is quite trivial: 
I want to transform the md before is added into the contenteditable div, specifically I want to replace all the ![...](...) and [...](...) tags with the  content that is inserted within the square brackets.
My current implementation is performing the transformation using a listener on the editor such as:

```
const editor = await Editor
  .make()
  .config(ctx => {
  ...
    ctx.get(listenerCtx)
      .markdownUpdated((ctx, markdown, prevMarkdown) => {
        if (markdown !== prevMarkdown) {
          const filteredMarkdown = markdown
          .replace(/!\[(.*?)\]\(.*?\)/g, '$1')
          .replace(/\[(.*?)\]\(.*?\)/g, '$1')

          ???
        }
     })
```

but then I dont know how to update the markdown with the filteredMarkdown, is this the default way ? any ideas?

**Comment by Saul-Mirone** [comment]
Are you looking for [replaceAll macro](https://milkdown.dev/docs/api/utils#replace-all-markdown-string-flush-boolean-false-fn-ctx-ctx)?
